plugins/peereval/main.go

```diff
@@ -0,0 +1,327 @@
+package peereval
+
+import (
+	"encoding/json"
+	"flag"
+	"fmt"
+	"sort"
+	"sync"
+	"time"
+
+	"github.com/Shopify/sarama"
+
+	gtypes "github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/ethereum/go-ethereum/node"
+	"github.com/ethereum/go-ethereum/rpc"
+
+	"github.com/openrelayxyz/xplugeth"
+	"github.com/openrelayxyz/xplugeth/hooks/fetcher"
+	"github.com/openrelayxyz/xplugeth/hooks/initialize"
+	"github.com/openrelayxyz/xplugeth/types"
+	"github.com/openrelayxyz/xplugeth/utils"
+)
+
+type peerEvalConfig struct {
+	BrokerURL string `yaml:"broker.url"`
+	Topic string `yaml:"topic"`
+}
+
+type PeerMetrics struct {
+	ID                string
+	Enode             string
+	BlocksContributed int
+	LastConnected     time.Time
+	LastDisconnected  time.Time
+	IsConnected       bool
+	ConnectedTime     time.Duration
+	IsInbound 		  bool 
+}
+
+type HealthyPeers interface {
+	GetHealthyPeers() []string
+}
+
+var (
+	blockCount = 0
+	client  *rpc.Client 
+	chainid            int64
+	cfg  *peerEvalConfig
+	flags                     = *flag.NewFlagSet("peereval-plugin", flag.ContinueOnError)
+	maxPeerCount              = flags.Int("peereval.max.peers", 0, "max peer value for peer eval plugin")
+	pollingInterval           = flags.Duration("peereval.polling.interval", time.Minute, "polling interval for peer monitoring")
+	connectionTimeCoefficient = flags.Duration("peereval.connection.time.coefficient", 5*time.Minute, "minimum connection time before evaluating peers")
+)
+
+type peerEvalModule struct {
+	peerMetricsMap map[string]*PeerMetrics
+	mutex          sync.Mutex
+	peerRatios     map[string]float64
+}
+
+func init() {
+	xplugeth.RegisterModule[peerEvalModule]("peerEvalModule")
+	xplugeth.RegisterHook[HealthyPeers]()
+	xplugeth.RegisterFlags(flags)
+}
+
+func (p *peerEvalModule) InitializeNode(s *node.Node, b types.Backend) {
+	client =  s.Attach()
+
+	config, ok := xplugeth.GetConfig[peerEvalConfig]("peereval")
+	if !ok {
+		log.Warn("peerEval config not found, using defaults")
+		cfg = &peerEvalConfig{}
+	} else {
+		cfg = config
+	}
+
+	if *maxPeerCount == 0 {
```

Comment:
You know you can just set the default on line 51 to something non-zero and it will get applied when parsed. That would preclude logging, but I generally think it's acceptable for flags to have default values and not log that they weren't specified.